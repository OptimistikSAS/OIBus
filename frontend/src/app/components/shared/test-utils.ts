import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
