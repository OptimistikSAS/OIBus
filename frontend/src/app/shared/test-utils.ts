import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
