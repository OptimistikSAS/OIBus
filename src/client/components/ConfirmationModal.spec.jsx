/**
 * @jest-environment jsdom
 */
import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

import ConfirmationModal from './ConfirmationModal.jsx'

import activeConfig from '../../../tests/testConfig'

const mockHistoryPush = jest.fn()

jest.mock('react-router-dom', () => (
  { useHistory: () => ({ push: mockHistoryPush }) }
))

let container

beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
  jest.clearAllMocks()
})

React.useContext = jest.fn().mockReturnValue({ activeConfig })
describe('ConfirmationModal', () => {
  test('display ConfirmationModal page based on config', async () => {
    act(() => {
      ReactDOM.render(
        <ConfirmationModal title="My Title" body="My Body" onConfirm={() => true} isOpen={false} toggle={() => true} />,
        container,
      )
    })
    expect(container).toMatchSnapshot()
  })
})
