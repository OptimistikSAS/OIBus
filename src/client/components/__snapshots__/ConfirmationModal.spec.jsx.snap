// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal display ConfirmationModal page based on config 1`] = `<div />`;
